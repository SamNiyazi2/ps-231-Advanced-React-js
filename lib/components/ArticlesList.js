
// 07/21/2020 04:22 am - SSN - [20200721-0002] - [003] - M03-P06 - Matching data to React components
// 07/21/2020 06:48 am - SSN - [20200721-0637] - [003] - M03-P08 - Components responsibility

import React from 'react';
import Article from './Article';

// 07/31/2020 09:13 pm - SSN - [20200731-0751] - [003] - M07-P03 - Perf addons: Avoiding wasteful renderers with PureComponent
// Functions always rerender, unless using HOC.  Replacing with class (PureComponent) for performance.

const ArticlesList_originalFunc = (props) => {
    return (
        <div>
            {Object.values(props.articles).map(article =>
                <Article
                    key={article.id}
                    article={article}

                // 07/23/2020 01:04 am - SSN - [20200722-1634] - [004] - M05-P02 - React's context API

                // store={props.store}
                />

            )}
        </div>
    );
};



class ArticlesList extends React.PureComponent {

    componentWillUpdate(nextProps, nextState) {
        console.log('ssn-20200731-2117 - Updating ArticlesList');
    }

    render() {
        return (
            <div>
                {Object.values(this.props.articles).map(article =>
                    <Article
                        key={article.id}
                        article={article}

                    />

                )}
            </div>
        )
    }
}




export default ArticlesList;
