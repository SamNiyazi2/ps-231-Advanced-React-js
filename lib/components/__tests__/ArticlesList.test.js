

// 07/21/2020 12:23 pm - SSN - [20200721-1205] - [001] - M03-P09 - Jest snapshot testing


import React from 'react';
import ArticlesList from '../ArticlesList';

import renderer from 'react-test-renderer';

describe('ArticlesList', () => {


    it('renders correctly', () => {

        const testProps = {
            articles: {
                a: {
                    id: 'a',
                    title: 'Title 101',
                    body: 'Body 101',
                    date: ''
                },
                b: {
                    id: 'b',
                    title: 'Title 102',
                    body: 'Body 102',
                    date: ''
                }
            }
            ,
            store: {
                lookupAuthor: jest.fn(() => { return {  firstName: 'SomeFirstName'}; })
            }

        };

        
        const testTree = renderer.create(
            <ArticlesList

                {...testProps}
            />

        ).toJSON();

        expect(testTree).toMatchSnapshot();

        

    });


});
