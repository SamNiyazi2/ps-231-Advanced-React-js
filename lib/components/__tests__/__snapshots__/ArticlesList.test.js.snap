// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlesList renders correctly 1`] = `
<div>
  <div>
    <div
      style={
        Object {
          "fontWeight": "bold",
          "paddingTop": 32,
        }
      }
    >
      Title 101
    </div>
    <div
      style={
        Object {
          "color": "#888",
          "fontSize": "0.8em",
        }
      }
    >
      Invalid Date
    </div>
    <div
      style={
        Object {
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      Author:  
      <a
        href="{author.website}"
      >
        SomeFirstName
         
      </a>
    </div>
    <div
      style={
        Object {
          "paddingLeft": 20,
        }
      }
    >
      Body 101
    </div>
  </div>
  <div>
    <div
      style={
        Object {
          "fontWeight": "bold",
          "paddingTop": 32,
        }
      }
    >
      Title 102
    </div>
    <div
      style={
        Object {
          "color": "#888",
          "fontSize": "0.8em",
        }
      }
    >
      Invalid Date
    </div>
    <div
      style={
        Object {
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      Author:  
      <a
        href="{author.website}"
      >
        SomeFirstName
         
      </a>
    </div>
    <div
      style={
        Object {
          "paddingLeft": 20,
        }
      }
    >
      Body 102
    </div>
  </div>
</div>
`;
